src/odemis/acq/move.py

```diff
@@ -875,40 +919,82 @@ def _doCryoSwitchSamplePosition(future, target):
                 raise CancelledError()
             future._task_state = FINISHED
 
-# Note: this transformation consists of translation of along x and y
-# axes, and 7 degrees rotation around rx, and 180 degree rotation around rz.
-# The rotation angles are constant existing in "FM_POS_ACTIVE" metadata,
-# but the translation are calculated based on the current position and some
-# correction/shifting parameters existing in metadata "FM_POS_ACTIVE". 
-# This correction parameters can change every session. They are calibrated
-# at the beginning of each run.
+
+def check_stage_metadata(calibrated_values: dict):
+    """
+    Check the required keys in the stage metadata based on given stage type in the microscope file.
+    :param calibrated_values: calibrated values to carry out stage transformation.
```

Comment:
I think it'd be nicer to either take the "stage" as input. The function can then decided which metadata to look at. You could even not take *any* hardware, and the function would itself call `.getComponent()` to get all the components that needs to be checked (ex: stage, stage-bare, focus).